kernel/thread/thread.c

```diff
@@ -475,6 +477,9 @@ kthread_t *thd_create_ex(const kthread_attr_t *restrict attr,
             /* Clear out potentially unused stuff */
             memset(nt, 0, sizeof(kthread_t));
 
+            /* Initialize the flags to defaults immediately. */
+            nt->flags = THD_DEFAULTS;
+
```

Comment:
This is good to have, even though it's already implicitly set by the memset above it and `THD_DEFAULTS` being 0. Really helps keep the reasoning from getting obscured.